iceoryx2/src/service/dynamic_config/event.rs

```diff
@@ -79,11 +79,19 @@ impl DynamicConfig {
         self.notifiers.len()
     }
 
-    pub(crate) fn add_listener_id(&self, id: UniqueListenerId) -> Option<UniqueIndex> {
-        unsafe { self.listeners.add(id) }
+    pub(crate) fn add_listener_id(&self, id: UniqueListenerId) -> Option<u32> {
+        unsafe { self.listeners.add_raw(id) }
     }
 
-    pub(crate) fn add_notifier_id(&self, id: UniqueNotifierId) -> Option<UniqueIndex> {
-        unsafe { self.notifiers.add(id) }
+    pub(crate) fn release_listener_id(&self, id: u32) {
+        unsafe { self.listeners.remove_raw_index(id) }
```

Comment:
`id` might be misleading here. It's the index of the container, right?

Maybe a `Handle` type could be created and used instead of the `u32` ... I just had a déjà vu. I think I already proposed a `Handle` somewhere else in this code base.

Not sure if the name should be kept or changed to e.g. `release_listener_id_with_handle`. The latter would leave the option to later re-introduce a method like `fn remove_listener_id(&self, id: UniqueListenerId)`